src/rgw/rgw_rest.cc

```diff
@@ -1351,14 +1314,17 @@ static int read_all_chunked_input(req_state *s, char **pdata, int *plen, int max
 }
 
 int rgw_rest_read_all_input(struct req_state *s, char **pdata, int *plen,
-			    int max_len)
+			    uint64_t max_len, bool allow_chunked)
 {
   size_t cl = 0;
   int len = 0;
   char *data = NULL;
 
   if (s->length)
     cl = atoll(s->length);
+  else if (!allow_chunked)
```

Comment:
Since we're using this as a guard, I'd prefer we checked it first and bailed earlier if possible. Note that the check that's coming up only occurs if "cl" isn't false and then does another check and maybe-exits-- I think this overall logic could be reworked into checking some conditions and returning or not early. (Or, I may need a cup of coffee!)